reframe/core/schedulers/firecrest.py

```diff
@@ -0,0 +1,364 @@
+        return _SlurmFirecrestJob(*args, **kwargs)
+
+    def _push_artefacts(self, job):
+        def _upload(local_path, remote_path):
+            f_size = os.path.getsize(local_path)
+            if f_size <= self._max_file_size_utilities:
+                self.client.simple_upload(
+                    self._system_name,
+                    local_path,
+                    remote_path
+                )
+            else:
+                self.log(
+                    f'File {f} is {f_size} bytes, so it may take some time...'
+                )
+                up_obj = self.client.external_upload(
+                    self._system_name,
+                    local_path,
+                    remote_path
+                )
+                up_obj.finish_upload()
+                return up_obj
+
+        for dirpath, dirnames, filenames in os.walk('.'):
+            for d in dirnames:
+                new_dir = join_and_normalize(job._remotedir, dirpath, d)
+                self.log(f'Creating remote directory {new_dir}')
+                self.client.mkdir(self._system_name, new_dir, p=True)
+
+            async_uploads = []
+            remote_dir_path = join_and_normalize(job._remotedir, dirpath)
+            for f in filenames:
+                local_norm_path = join_and_normalize(
+                    job._localdir, dirpath, f
+                )
+                modtime = os.path.getmtime(local_norm_path)
+                last_modtime = self._local_filetimestamps.get(local_norm_path)
+                if (last_modtime != modtime):
+                    self._local_filetimestamps[local_norm_path] = modtime
+                    self.log(
+                        f'Uploading file {f} in '
+                        f'{join_and_normalize(job._remotedir, dirpath)}'
+                    )
+                    up = _upload(
+                        local_norm_path,
+                        remote_dir_path
+                    )
+                    if up:
+                        async_uploads.append(up)
+
+            sleep_time = itertools.cycle([1, 5, 10])
+            while async_uploads:
+                still_uploading = []
+                for element in async_uploads:
+                    upload_status = int(element.status)
+                    if upload_status < 114:
+                        still_uploading.append(element)
+                        self.log(f'file is still uploafing, '
+                                 f'status: {upload_status}')
+                    elif upload_status > 114:
+                        raise JobSchedulerError(
+                            'could not upload file to remote staging '
+                            'area'
+                        )
+
+                async_uploads = still_uploading
+                t = next(sleep_time)
+                self.log(
+                    f'Waiting for the uploads, sleeping for {t} sec'
+                )
+                time.sleep(t)
```

Comment:
I think my previous comment about the framework responsiveness makes no sense, right? It will sleep here.